src/util/worker/mod.rs

```diff
@@ -193,6 +239,22 @@ impl<T: Display + Send + 'static> Worker<T> {
             scheduler: Scheduler::new(name, AtomicUsize::new(0), tx),
             receiver: Mutex::new(Some(rx)),
             handle: None,
+            tick_duration: Duration::from_secs(NAP_SECS),
+            tasks_per_tick: DEFAULT_TASKS_PER_TICK,
+        }
+    }
+
+    /// Set the `tick_duration` of the worker, must be greater than 0.
```

Comment:
I suggest passing an Option to the worker creator or using a factory class to build the work.